tests/Integration/Behaviour/Features/Context/Domain/CurrencyFeatureContext.php

```diff
@@ -0,0 +1,160 @@
+            (bool) $data['is_enabled']
+        );
+
+        $command->setShopIds([
+            (int) $shop->id,
+        ]);
+
+        /** @var CurrencyId $currencyId */
+        $currencyId = $this->getCommandBus()->handle($command);
+
+        SharedStorage::getStorage()->set($reference, new Currency($currencyId->getValue()));
+    }
+
+    /**
+     * @When I edit currency :reference with following properties:
+     */
+    public function editCurrency($reference, TableNode $node)
+    {
+        $data = $node->getRowsHash();
+        /** @var Currency $currency */
+        $currency = SharedStorage::getStorage()->get($reference);
+
+        $command = new EditCurrencyCommand((int) $currency->id);
+
+        if (isset($data['iso_code'])) {
+            $command->setIsoCode($data['iso_code']);
+        }
+
+        if (isset($data['exchange_rate'])) {
+            $command->setExchangeRate((float) $data['exchange_rate']);
+        }
+
+        if (isset($data['is_enabled'])) {
+            $command->setIsEnabled((bool) $data['is_enabled']);
+        }
+
+        if (isset($data['shop_association'])) {
+            $command->setShopIds([(int) $data['shop_association']]);
+        }
+
+        $this->getCommandBus()->handle($command);
+
+        SharedStorage::getStorage()->set($reference, new Currency($currency->id));
+    }
+
+    /**
+     * @When I disable currency ":currencyReference"
+     */
+    public function disableCurrency($reference)
+    {
+        /** @var Currency $currency */
+        $currency = SharedStorage::getStorage()->get($reference);
+
+        try {
+            $this->getCommandBus()->handle(new ToggleCurrencyStatusCommand((int) $currency->id));
+        } catch (CannotDisableDefaultCurrencyException $e) {
+            $this->lastException = $e;
+        }
+    }
+
+    /**
+     * @When I delete currency ":currencyReference"
+     */
+    public function deleteCurrency($reference)
+    {
+        /** @var Currency $currency */
+        $currency = SharedStorage::getStorage()->get($reference);
+
+        try {
+            $this->getCommandBus()->handle(new DeleteCurrencyCommand((int) $currency->id));
+        } catch (CannotDeleteDefaultCurrencyException $e) {
+            $this->lastException = $e;
+        }
+    }
+
+    /**
+     * @Then I should get error that default currency cannot be disabled
+     */
+    public function assertLastErrorIsDefaultCurrencyCannotBeDisabled()
+    {
+        if (!$this->lastException instanceof CannotDisableDefaultCurrencyException) {
+            throw new RuntimeException(sprintf(
+                'Last error should be "CannotDisableDefaultCurrencyException", but got "%s"',
+                $this->lastException ? get_class($this->lastException) : 'null'
+            ));
+        }
+    }
+
+    /**
+     * @Then I should get error that default currency cannot be deleted
+     */
+    public function assertLastErrorIsDefaultCurrencyCannotBeDeleted()
```

Comment:
Looks a lot like `assertLastErrorIsDefaultCurrencyCannotBeDisabled` 🤔what about we create generic `assertLastErrorIs($exceptionClass)` ?

Reply:
If you mean adding exception class to behat step, like:

`@Then I should get "Prestashop\Exception\ErrorException" exception`

Then I don't think it's a good idea as it makes behat steps very technical. Or did I understand it wrong?

Reply:
You're right, steps should stay human-oriented.
I was thinking of either
```
/**
 * @Then I should get error that default currency cannot be deleted
 */
public function assertLastErrorIsDefaultCurrencyCannotBeDeleted()
{
    $this->assertLastErrorIs(CannotDeleteDefaultCurrencyException::class);
}
```
or
```
/**
 * @Then I should get error that "(.*)"
 */
public function assertLastErrorIs($errorType)
{
    $knownErrorTypes = [
        'default currency cannot be deleted' => CannotDeleteDefaultCurrencyException::class,
        'default currency cannot be disabled' => CannotDisableDefaultCurrencyException::class,
    ];

    $exceptionClass = $knownErrorTypes[$errorType];

    $this->assertLastErrorIs($exceptionClass);
}
```

Reply:
Done!